compiler/src/tests/cxxfrontend.cc

```diff
@@ -1172,12 +1172,6 @@ class MiniGlueVisitor : public Visitor
                 Dsymbol *s = (*d->members)[i];
                 s->accept(this);
             }
-            ClassDeclarations aclasses;
-            for (size_t i = 0; i < d->members->length; i++)
-            {
-                Dsymbol *member = (*d->members)[i];
-                member->addLocalClass(&aclasses);
-            }
```

Comment:
This needs a suitable replacement in C++ too.

Reply:
Is it ever used from C++?

Reply:
The only use for this in D is to add info to the ModuleInfo, which is being deprecated.

Reply:
This is part of the code generator, so GDC and LDC need to generate this information somehow (both written in C++).  If you're going to break downstream compilers there should at least be a migration path for us to do the equivalent.

Reply:
@ibuclaw is it used extensively? If it's just one occurrence as is the case in this PR, maybe it can just be fixed?

Reply:
Update how? The replacement is `extern (D)` !!!

Reply:
The array of class declarations is extern(C++), `AttribDeclaration.include` is extern(C++), the only function that is not extern(C++) is `foreachDsymbol` but that could be made extern(C++).

I'm not saying this is the best approach, I'm just trying to understand what the best option is.

Reply:
Maybe....
```
class Module
{
    extern(C++) final void addLocalClasses(ref ClassDeclarations aclasses)
   {
        //d-specific implementation using delegates and _foreach
   }
}
```
Whoever is modifying the backend/frontend interface needs to think about this.

Reply:
done